markhor_flippers/src/markhor_flippers.cpp

```diff
@@ -32,40 +38,184 @@ void joyCallback(const sensor_msgs::Joy::ConstPtr& joy)
     flipper_rl_pub.publish(msg);
     flipper_rr_pub.publish(msg);
   }
+  if (individual_mode == true)
+  {
+    ROS_INFO("individual mode is successful"); 
+
+    if (flipper_mode_fl == true)
+    {
+      msg.data = joy->axes[4] * multiplicator;
+      flipper_fl_pub.publish(msg);
+    }
+    if (flipper_mode_fr == true)
+    {
+      msg.data = joy->axes[4] * multiplicator;
+      flipper_fr_pub.publish(msg);
+    }
+    if (flipper_mode_rl == true)
+    {
+      msg.data = joy->axes[4] * multiplicator;
+      flipper_rl_pub.publish(msg);
+    }
+    if (flipper_mode_rr == true)
+    {
+      msg.data = joy->axes[4] * multiplicator;
+      flipper_rr_pub.publish(msg);
+    }
+  }
 }
 
 bool flipperModeFrontEnable(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& res)
 {
   flipper_mode_front = true;
-  res.message = "successfully enable flipper mode front";
+  res.message = "successfully enabled flipper mode front";
   res.success = static_cast<unsigned char>(true);
   return true;
 }
 
 bool flipperModeFrontDisable(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& res)
 {
   flipper_mode_front = false;
-  res.message = "successfully disable flipper mode front";
+  res.message = "successfully disabled flipper mode front";
   res.success = static_cast<unsigned char>(true);
   return true;
 }
 
 bool flipperModeBackEnable(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& res)
 {
   flipper_mode_back = true;
-  res.message = "successfully enable flipper mode back";
+  res.message = "successfully enabled flipper mode back";
   res.success = static_cast<unsigned char>(true);
   return true;
 }
 
 bool flipperModeBackDisable(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& res)
 {
   flipper_mode_back = false;
-  res.message = "successfully disable flipper mode back";
+  res.message = "successfully disabled flipper mode back";
+  res.success = static_cast<unsigned char>(true);
+  return true;
+}
+
+bool diagnosticModeEnable(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& res)
+{
+  individual_mode = true;
+  flipper_mode_front = false;
+  flipper_mode_back = false;
+  res.message = "successfully enabled individual mode";
+  res.success = static_cast<unsigned char>(true);
+  return true;
+}
+
+bool diagnosticModeDisable(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& res)
+{
+  individual_mode = false;
+  flipper_mode_fl = false;
+  flipper_mode_fr = false;
+  flipper_mode_rl = false;
+  flipper_mode_rr = false;
+
+  res.message = "successfully disabled individual mode";
   res.success = static_cast<unsigned char>(true);
   return true;
 }
 
+bool flipperModeFLEnable(std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& res)
```

Comment:
Also, this is a fine implementation (using 4 services). But the thing is that there's a lot of repetition in the code, ideally, instead of using a trigger service, we would have a specific message that tells which flippers we are calling and remove the redundant code.

I know you could do the same arguments from what I did with the flipper mode, and that would be a fair argument. I'm simply saying this because I want you guys to write the best code I can think of when you have the time to do it. 

Ask me for clarification if you're not sure about what I'm saying.

Reply:
Do you mean combining the services into one and using handles, as mentioned here: [http://wiki.ros.org/roscpp/Overview/Services](url) ?

Reply:
Humm not really, I'm more talking about doing our own service message and in that message including the 4 flippers status (enable or disable). And inside a single function, check and enable/disable the flippers. 

Something similar to this : http://wiki.ros.org/ROS/Tutorials/CreatingMsgAndSrv#Using_rossrv

It will reduce the code base and simply the design a bit.

Reply:
But that can be done some other time. It's more to give you a head's up.